commit aaff41001fdf4d8a7f8425dc5655f65a1ccb3344
Author: David Kilzer <ddkilzer@apple.com>
Date:   Tue Jan 26 16:57:03 2016 -0800

    Heap-based buffer-underreads due to xmlParseName
    
    For https://bugzilla.gnome.org/show_bug.cgi?id=759573
    
    * parser.c:
    (xmlParseElementDecl): Return early on invalid input to fix
    non-minimized test case (759573-2.xml).  Otherwise the parser
    gets into a bad state in SKIP(3) at the end of the function.
    (xmlParseConditionalSections): Halt parsing when hitting invalid
    input that would otherwise caused xmlParserHandlePEReference()
    to recurse unexpectedly.  This fixes the minimized test case
    (759573.xml).
    
    Had to fixup by removing tests as they failed on RHEL-6

diff --git a/parser.c b/parser.c
index 20002d4..aa8b63a 100644
--- a/parser.c
+++ b/parser.c
@@ -6499,6 +6499,7 @@ xmlParseElementDecl(xmlParserCtxtPtr ctxt) {
 	if (!IS_BLANK_CH(CUR)) {
 	    xmlFatalErrMsg(ctxt, XML_ERR_SPACE_REQUIRED,
 		           "Space required after 'ELEMENT'\n");
+	    return(-1);
 	}
         SKIP_BLANKS;
         name = xmlParseName(ctxt);
@@ -6650,6 +6651,7 @@ xmlParseConditionalSections(xmlParserCtxtPtr ctxt) {
 
 	    if ((CUR_PTR == check) && (cons == ctxt->input->consumed)) {
 		xmlFatalErr(ctxt, XML_ERR_EXT_SUBSET_NOT_FINISHED, NULL);
+		xmlHaltParser(ctxt);
 		break;
 	    }
 	}
